### DISCUSSION TITLE
Reason isn't displayed when stop workflow with CANCEL stoptype

### DISCUSSION THREAD

**Discussion by nakanaori** [indeedeng/iwf#329] [Q&A]
When we stop the workflow using FAIL StopType, the reason that we specify is shown. Why do When we use CANCEL StopType, the reason that we specify isn't shown? 

both of these use StopWorkflow function from IWF Client

- FAIL stoptype
![image](https://github.com/indeedeng/iwf/assets/37839775/2a2a66a3-290f-4ddc-b357-10dfe2defb43)

- CANCEL stoptype
![image](https://github.com/indeedeng/iwf/assets/37839775/23b48467-febc-4eec-b86c-b159a1868927)

**Comment by longquanzheng** [comment]
This is because Temporal SDK doesn't support this: https://github.com/temporalio/sdk-go/blob/7dbde86f46b75cd99dc8e36cedc1bfd45e2bc01e/client/client.go#L299

It doesn't provide a way to pass a reason. 

As a result, iWF can not pass it to the CancelEvent: https://github.com/indeedeng/iwf/blob/696223a3acd6c2d909d74b18e6f4a3021fda77e1/service/api/service.go#L177

There is a related problem for Terminate stop type. Even thought terminate allow passing reason, the get workflow result API of Temporal client doesn't allow retrieving the reason back..I just opened an issue/feature request: https://github.com/temporalio/features/issues/325

**Reply by longquanzheng** [reply]
@nakanaori how important is that for you? I just think out a workaround if you want to just persist the message so that it won’t be lost .

**Reply by nakanaori** [reply]
Thank you for replying @longquanzheng. Do you mean we persist the reason in our own database?

**Reply by longquanzheng** [reply]
No. I am thinking that in iWF API service, right before this line, https://github.com/indeedeng/iwf/blob/696223a3acd6c2d909d74b18e6f4a3021fda77e1/service/api/service.go#L177 , send a signal to the workflow with a iWF system signal name https://github.com/indeedeng/iwf/blob/696223a3acd6c2d909d74b18e6f4a3021fda77e1/service/const.go#L35 , like 
```golang
StopReasonSignalChannelName    = IwfSystemConstPrefix + "StopReason"
```
So that the reason will be recorded into history. 

It will be just a few lines of code in iWF.
